lib/ansible/plugins/lookup/config.py

```diff
@@ -154,13 +147,11 @@ def run(self, terms, variables=None, **kwargs):
                     result, origin = _get_plugin_config(pname, ptype, term, variables)
                 else:
                     result = _get_global_config(term)
-            except MissingSetting as e:
+            except AnsibleOptionsError as e:
                 if missing == 'error':
-                    raise AnsibleLookupError('Unable to find setting %s' % term, orig_exc=e)
-                elif missing == 'warn':
-                    self._display.warning('Skipping, did not find setting %s' % term)
-                elif missing == 'skip':
-                    pass  # this is not needed, but added to have all 3 options stated
+                    raise AnsibleLookupError(f'Unable to find setting {term}', orig_exc=e)
+                if missing == 'warn':
+                    self._display.warning(f'Skipping, did not find setting {term}')
```

Comment:
Since you are removing the missing check above, this should assume `missing == 'error'` as an else but also declare `warn` and `skip`
```
                if missing == 'warn':
                    self._display.warning('Skipping, did not find setting %s' % term)
                elif missing != 'skip':
                    raise AnsibleLookupError('Unable to find setting %s' % term, orig_exc=e)
```